### PR TITLE
minor simplification of v3.1 schema

### PR DESCRIPTION
**Pull request by karenetheridge**
"if the X property is present, then apply this constraint to property X" can
be simplified by simply stating the constraint for property X.

### CODE REVIEW

**Review by jdesrosiers — CHANGES_REQUESTED** [review]
You're right, this doesn't work like I intended.

Would you mind also making a couple other corrections I noticed when reviewing?

1. The "allowEmptyValue" property doesn't apply to headers. That can be removed.
2. The schema doesn't include the constraint that one and only of "schema" or "content" is required.

**Comment by karenetheridge** [comment]
yes, give me a day or two

**Comment by karenetheridge** [comment]
> The "allowEmptyValue" property doesn't apply to headers. That can be removed.

Are you sure? https://spec.openapis.org/oas/v3.1.0#headerObject specifies that it's the same as parameter, except no 'name' or 'in' -- and 'parameter' has 'allowEmptyValue'.

> The schema doesn't include the constraint that one and only of "schema" or "content" is required.

done.

**Comment by karenetheridge** [comment]
since the header definition has 'unevaluatedProperties: false' in it, does this mean that no one has been testing this schema yet against headers (or surely they would have noticed the allowEmptyValue discrepancy)??!

**Comment by jdesrosiers** [comment]
> > The "allowEmptyValue" property doesn't apply to headers. That can be removed.
>
> Are you sure?

Sorry, I meant the "allowReserved" property doesn't apply to headers. :facepalm: 

I came to this conclusion because the `in` value is implicitly `header` and the `allowReserved` description says, "This property only applies to parameters with an `in` value of `query`." Therefore, I concluded that `allowReserved` doesn't apply to the Header Object.

Actually, now that I look at it again, it looks like the same is true of `allowEmptyValue`. It's description says, "This is valid only for `query` parameters". This is for a header parameter, so it doesn't apply either.

File: schemas/v3.1/schema.yaml [hunk 1]
```diff
@@ -650,13 +656,7 @@ $defs:
           allowReserved:
             default: false
             type: boolean
```
Comment (jdesrosiers) [inline schemas/v3.1/schema.yaml:658]: Let's remove this one too.
```suggestion
```

**Review by jdesrosiers — CHANGES_REQUESTED** [review]

File: schemas/v3.1/schema.yaml (hunk 1 above)
Comment (karenetheridge) [reply schemas/v3.1/schema.yaml:658]: done.

**Review by jdesrosiers — APPROVED** [review]
Thanks!

**Review by jdesrosiers — APPROVED** [review]

**Review by darrelmiller — APPROVED** [review]

**Pull request merged**